src/main/java/com/googlecode/download/maven/plugin/internal/cache/FileBackedIndex.java

```diff
@@ -19,50 +43,124 @@
  * @author Paul Polishchuk
  * @since 1.3.1
  */
-@NotThreadSafe
-final class FileBackedIndex implements FileIndex {
+@ThreadSafe
+public class FileBackedIndex implements HttpCacheStorage {
+
+    private static final Pattern URI_REGEX = Pattern.compile("^(?:\\{.*})?([^/]+//?.*)$");
     private static final String CACHE_FILENAME = "index.ser";
-    private final Map<URI, String> index = new HashMap<>();
-    private final File storage;
-    private final ReentrantLock lock = new ReentrantLock();
+    private final Map<URI, String> index = new ConcurrentHashMap<>();
+    private final Path cacheIndexFile;
     private final Log log;
+    private final Path baseDir;
+
+    static class HttpCacheEntryAdapter {
```

Comment:
@ajarmoniuk can't methods of the class be private static methods of  FileBackedIndex ?